src/ranged.cpp

```diff
@@ -1766,17 +1766,8 @@ double player::gun_value( const item &weap, int ammo ) const
 
     damage_instance gun_damage = weap.gun_damage();
     item tmp = weap;
-    if( tmp.is_magazine() ) {
+    if( tmp.is_magazine() || tmp.magazine_current() || tmp.magazine_default() ) {
```

Comment:
you sure about this? ammo_set() does not currently handle setting a magazine item and then setting ammo inside an item.

Reply:
I double checked `ammo_set`. As the comment says:
```cpp
/**
 * Filter setting the ammo for this instance
 * Any existing ammo is removed. If necessary a magazine is also added.
 * @param ammo specific type of ammo (must be compatible with item ammo type)
 * @param qty maximum ammo (capped by item capacity) or negative to fill to capacity
 * @return same instance to allow method chaining
 */
item &ammo_set( const itype_id &ammo, int qty = -1 );
```
This function really if necessary creates a magazine and inserts it into an item, then fills it with ammo.
```cpp
// ...
if( !magazine_current() ) {
	const itype *mag = find_type( magazine_default() );
	// ...
	put_in( item( mag ), item_pocket::pocket_type::MAGAZINE_WELL );
}
item *mag_cur = magazine_current();
if( mag_cur != nullptr ) {
    mag_cur->ammo_set( ammo, qty );
}
```